activesupport/lib/active_support/logger.rb

```diff
@@ -46,6 +46,31 @@ def initialize(*args)
       @formatter = SimpleFormatter.new
     end
 
+    def level
+      # The log level should initially be setup in the main thread during startup,
+      # which means that if our current thread doesn't have a local level set, then
+      # we attempt to use the main thread's level, falling back to the class default
+      # otherwise.
+      Thread.current[:logger_level] || Thread.main[:logger_level] || super
+    end
+
+    def level=(level)
+      Thread.current[:logger_level] = level
+    end
+
+    def add(severity, message = nil, progname = nil, &block)
+      return true if @logdev.nil? || (severity || UNKNOWN) < level
```

Comment:
ActiveSupport::Logger inherits from Logger, so it does have "control" over ``@logdev``. The reason it's checked here is because it's replicating the behavior of the super method. I personally see no problem short-circuiting here instead of adding another method call to the stack (via ``super``) but, if you feel super strongly about it, I'll be honest - I don't care, and will remove it.